### PR TITLE
add two tests for ligo compile file handling

### LINKED ISSUE #1815: 🚧 Dev ➾ add an E2E test for ligo compile with misnamed parameter file
### 🔗 Epic|Feature Link

_No response_

### 🚥 Definition of Done

- [ ] 🎯 Planning and Design
- [ ] 🏎️ Dev Implementation Complete
- [ ] ♾️ PR Merged
- [ ] 🏆 Product Acceptance

### 🧰 Work Description

Apparently, when ligo is asked to compile a contract, but the parameter file is misnamed r was renamed it does not work.
Make a test to show this failure so that when it is fixed it can be tested/

Initialize a new project: taq init test-project
Navigate to the project: cd test-project
Install ligo plugin: taq install ../taqueria/taqueria-plugin-ligo
Copy ligo source files from tests: cp ../taqueria/tests/e2e/data/ligo-data/* contracts/
Rename parameters file: mv contracts/hello-tacos.parameters.mligo contracts/hello-tacos.parameterList.mligo
Compile: taq compile hello-tacos.mligo
Result:
```
┌─────────────────────────────────┬──────────────────────────┐
│ Contract                        │ Artifact                 │
├─────────────────────────────────┼──────────────────────────┤
│ hello-tacos.mligo               │ artifacts/hello-tacos.tz │
├─────────────────────────────────┼──────────────────────────┤
│ hello-tacos.storageList.mligo   │                          │
├─────────────────────────────────┼──────────────────────────┤
│ hello-tacos.parameterList.mligo │                          │
```
### ✅ To Do

make tests

### ⚖️ Acceptance Criteria

✨ Functional: 
- [ ] 
- [ ] working tests

### 🔭 Scope

🙅‍♂️ Out of Scope: 
- ... 
 
 tests of other plugins

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by michaelkernaghan**
# 🌮 Taqueria PR
closes #1815 
## 🪁 Description

add two tests for ligo compile file handling

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [x ] ⛱️ I have completed this PR template in full and updated the title appropriately
- [ ] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] 🐬 I have commented my code, particularly in hard-to-understand areas
- [ ] 🤿 Corresponding changes have been made to all documentation
- [ ] 🐚 Required changes to the VScE have been made
- [ ] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

add two tests for ligo compile file handling

## 🎢 Test Plan

add two tests for ligo compile file handling

## 🛸 Type of Change

- [x ] 🧽 Chore ➾
- [ ] 🛠️ Fix ➾
- [ ] ✨ Feature ➾
- [ ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

File: tests/e2e/ligo-plugin-e2e-tests.spec.ts [hunk 1]
```diff
@@ -237,4 +237,80 @@ describe('Ligo Plugin E2E Testing for Taqueria CLI', () => {
...
+                "├───────────────────────────┼───────────────────────────────────────────────┤",
+                "│ counter.storageList.mligo │ artifacts/counter.default_storage.tz          │",
+                "│                           │ artifacts/counter.storage.another_count.tz    │",
+                "├───────────────────────────┼───────────────────────────────────────────────┤",
+                "│ counter.parameters.mligo  │ artifacts/counter.parameter.increment_by_3.tz │",
+                "└───────────────────────────┴───────────────────────────────────────────────┘",
+              ]
+          `);
+
+		await execute('taq', 'install ../taqueria-plugin-flextesa', './test-project');
```
Comment (mweichert) [inline tests/e2e/ligo-plugin-e2e-tests.spec.ts:275]: I think lines 275-through-283 can be removed right? They don't seem relevant to this test.

File: tests/e2e/ligo-plugin-e2e-tests.spec.ts [hunk 2]
```diff
@@ -237,4 +237,80 @@ describe('Ligo Plugin E2E Testing for Taqueria CLI', () => {
...
+		await writeFile('./test-project/contracts/hello-tacos.mligo', mligo_file);
+		const parameters_file = await (await exec(`cat e2e/data/ligo-data/hello-tacos.parameters.mligo`)).stdout;
+		await writeFile('./test-project/contracts/hello-tacos.parameters.mligo', parameters_file);
+		expect(await ls('./test-project/contracts')).toEqual(
+			expect.arrayContaining(['hello-tacos.mligo', 'hello-tacos.parameters.mligo']),
+		);
+
+		await execute('taq', 'compile hello-tacos.mligo', './test-project');
+		expect.arrayContaining(['│ hello-tacos.parameters.mligo │                          │']),
+			await execute('taq', 'install ../taqueria-plugin-flextesa', './test-project');
```
Comment (mweichert) [inline tests/e2e/ligo-plugin-e2e-tests.spec.ts:305]: Lines 305-through-313 can be removed, right?

**Review by mweichert — CHANGES_REQUESTED** [review]
Looks good! Thanks Michael.

There were some lines in the tests that I wasn't sure about - they looked to have been results of a copy/paste. Do you mind checking whether they're deliberately included?

**Comment by michaelkernaghan** [comment]
> ligo-plugin-e2e-tests.spec.ts

thanks, those were lines that i neglected to delete, thanks

File: tests/e2e/ligo-plugin-e2e-tests.spec.ts (hunk 2 above)
Comment (michaelkernaghan) [reply tests/e2e/ligo-plugin-e2e-tests.spec.ts:305]: removed and correct replacement made

thanks

**Review by mweichert — APPROVED** [review]
